src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -551,7 +599,24 @@ private IVariableInformation[] ExpandVisualized(IVariableInformation variable)
                             arrayExpr.EnsureChildren();
                             if (arrayExpr.CountChildren != 0)
                             {
-                                children.AddRange(arrayExpr.Children);
+                                uint currentIndex = 0;
+                                var pvwVariable = variable as PaginatedVisualizerWrapper;
+                                if (pvwVariable != null)
+                                {
+                                    currentIndex = (variable as PaginatedVisualizerWrapper).StartIndex;
+                                }
+                                uint maxIndex = currentIndex + MAX_EXPAND > size ? size : currentIndex + MAX_EXPAND;
+                                for (uint index = currentIndex; index < maxIndex; ++index)
+                                {
+                                    children.Add(arrayExpr.Children[index]);
```

Comment:
I would be careful here.

Can you try a natvis where the actual number of children in the array is like 5 but the user hard-codes it to be 500?
We could be getting an indexOutOfBoundsException

Reply:
May you elaborate on this? If the user hardcodes the array to be of size 500, how would the number of children in the array be 5? Unless you mean we have an array `int arr [5]` and the user tries to access `arr[500]`?

Reply:
Change the SimpleVector definition in Natvis to hardcode a size outside of the actual code size.
```
  <Type Name="SimpleVector">
   <DisplayString>{{ size={_size} }}</DisplayString>
   <Expand>
    <Item Name="Size">_size</Item>
    <ArrayItems>
      <Size>500</Size>
      <ValuePointer>_start</ValuePointer>
    </ArrayItems>
   </Expand>
  </Type>
```

We should only be displaying values from valid indexes.

Reply:
Actually this seems to be ok. The internal IVariableInformation allocates enough space for Children that we will not be indexing out of bounds.